src/app/packages/map-facade/components/context-menu/context-menu.component.spec.ts

```diff
@@ -1,40 +1,162 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
-
+import { async, ComponentFixture, inject, TestBed } from '@angular/core/testing';
 import { ContextMenuComponent } from './context-menu.component';
 import { FormsModule } from '@angular/forms';
+import { Store, StoreModule } from '@ngrx/store';
+import { MapReducer } from '../../reducers/map.reducer';
+import { MapEffects } from '../../effects/map.effects';
+import { EventEmitter } from '@angular/core';
+import { ContextMenuDisplayAction, ContextMenuShowAction } from '../../actions/map.actions';
+import { EffectsTestingModule } from '@ngrx/effects/testing';
 
 describe('ContextMenuComponent', () => {
 	let component: ContextMenuComponent;
 	let fixture: ComponentFixture<ContextMenuComponent>;
+	let store: Store<any>;
+	const mockMapEffects = {onContextMenuShow$: new EventEmitter<void>()};
 
 	beforeEach(async(() => {
 		TestBed.configureTestingModule({
-			imports: [FormsModule],
-			declarations: [ ContextMenuComponent ]
-		})
-			.compileComponents();
+			imports: [FormsModule, EffectsTestingModule, StoreModule.provideStore({map: MapReducer})],
+			declarations: [ ContextMenuComponent ],
+			providers: [{provide: MapEffects, useValue: mockMapEffects}]
+		}).compileComponents();
 	}));
 
-	beforeEach(() => {
+	beforeEach(inject([Store], (_store: Store<any>) => {
 		fixture = TestBed.createComponent(ContextMenuComponent);
 		component = fixture.componentInstance;
 		fixture.detectChanges();
-	});
+		store = _store;
+	}));
 
 	it('should be created', () => {
 		expect(component).toBeTruthy();
 	});
 
-	it('get contextMenuStyle should return object with {top, left} from input', () => {
-		component.top = 20;
-		component.left = 10;
+	it('tabindex attribute should return zero', () => {
+		expect(component.tabindex).toEqual(0);
+	});
+
+	it('onmousewheel should get hide function', () => {
+		spyOn(component, 'hide');
+		component.onmousewheel();
+		expect(component.hide).toHaveBeenCalled();
+	});
+
+	it('contextmenu should call preventDefault on event arg', () => {
+		const $event = jasmine.createSpyObj({preventDefault: () => {}});
+		component.contextmenu($event);
+		expect($event.preventDefault).toHaveBeenCalled();
+	});
+
+	it('hide should call to blur on host element', () => {
+		spyOn(fixture.nativeElement, 'blur');
+		component.hide();
+		expect(fixture.nativeElement.blur).toHaveBeenCalled();
+	});
+
+	it('show should: save action, set {top, left} on host element(by renderer), call focus(host element) and initializeSensors', () => {
+		const elem = fixture.nativeElement;
+		spyOn(elem, 'focus');
+		spyOn(component, 'initializeSensors');
+		const action = <ContextMenuShowAction> {payload: {e: {x: 1, y: 2}}};
```

Comment:
Check that values are different before calling 'show'
expect(elem.style.top).not().toEqual('2px');
expect(elem.style.left).not().toEqual('1px');